.github/workflows/docs.yml

```diff
@@ -0,0 +1,25 @@
+name: Docs
+
+on: [push, pull_request]
+
+jobs:
+    docs:
+        name: Auto generate docs
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v4
+
+            - uses: kdheepak/panvimdoc@main
+              with:
+                vimdoc: harpoon
+                pandoc: README.md
+                description: Best plugin ever created
+                demojify: true
+                treesitter: true
+                shiftheadinglevelby: -1 
```

Comment:
@willothy this step generates the vimdoc from the README.md file.
which requires writing the markdown in a way that will be better converted to the vimdoc

From the comments above I can propose these options:
1. create another markdown file e.g. DOC.md which will contain documentation
2. update README.md to support seamless conversion to  vimdoc
3. write docs in the lua files and generate docs from there as it is done in https://github.com/nvim-telescope/telescope.nvim/blob/master/scripts/gendocs.lua

Reply:
Yeah, I guess it really depends on what prime prefers, but imo the simplest solution would be to just update the readme a bit. Overall though this is a great PR and I think autogenerated docs will be super helpful for everyone :)

Reply:
so the next course of action is to wait for @ThePrimeagen's response, right?

Reply:
Yep :)